### PR TITLE
Add stack trace to all current suspensions on early exit

### PR DESCRIPTION
**Pull request by kelunik**
Example output: https://gist.github.com/kelunik/eb5649f872755960a4f53b312d8f00fd

### CODE REVIEW

File: src/EventLoop/Internal/DriverSuspension.php [hunk 1]
```diff
@@ -87,7 +90,22 @@ public function suspend(): mixed
...
+            if ($suspensions) {
+                \gc_collect_cycles();
+
+                foreach ($suspensions as $suspension) {
+                    if ($suspension->fiber === null) {
+                        continue;
+                    }
+
+                    $reflectionFiber = new \ReflectionFiber($suspension->fiber);
+                    $info .= "\n\n" . $this->formatStacktrace($reflectionFiber->getTrace(\DEBUG_BACKTRACE_IGNORE_ARGS));
```
Comment (bwoebi) [inline src/EventLoop/Internal/DriverSuspension.php:104]: possibly check `ini_get("zend.exception_ignore_args")`? I like args :-D

**Comment by bwoebi** [comment]
This is a great idea, thanks for this.

File: src/EventLoop/Internal/DriverSuspension.php (hunk 1 above)
Comment (kelunik) [reply src/EventLoop/Internal/DriverSuspension.php:104]: Yes, I actually dumped the args locally, so I'll improve this.

**Comment by WyriHaximus** [comment]
So to be clear, this only happens when the event loop ends while there are still fibers around with potential work? And this visualise each stack trace for those fibers?

**Comment by kelunik** [comment]
Yup

**Review by WyriHaximus — APPROVED** [review]
Neat :+1:

**Review by trowski — APPROVED** [review]
Very cool! 😎

File: src/EventLoop/Internal/DriverSuspension.php (hunk 1 above)
Comment (kelunik) [reply src/EventLoop/Internal/DriverSuspension.php:104]: I'll work on this once needed.

**Pull request merged**